testdata/rabbit/write2disk/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"os"
+	"strings"
+)
+
+func main() {
+	message := "Go is expressive, concise, clean, and efficient. " +
+		"Its concurrency mechanisms make it easy to write programs" +
+		"that get the most out of multicore and networked machines, " +
+		"while its novel type system enables flexible and modular program construction. " +
+		"Go compiles quickly to machine code yet has the convenience of garbage" +
+		"collection and the power of run-time reflection."
+	// change the repeat count value to write longer data
+	longString := strings.Repeat(message, 1000000)
+	data := []byte(longString)
+
+	// try to write 2 files
+	// change this to write an unlimited number of files
+	for i := 0; i < 2; i++ {
+		writeDate(data)
+	}
+
+}
+
+func writeDate(data []byte) {
```

Comment:
The error handling in go is kinda weird.  For this small of a program, your use of a "check" function is ok.

As programs get larger, though, it is customary to have each layer of the call stack check the lower layer.  Example:

```// level 1
func foo() (*Fleazil, error) {
   fl, err:=bar()
   if err!=nil {
     return nil, err
  }
 // do something with the fleazil
// ...
}

// level 2
func bar() (*Fleazil, error) {
  fl, err := baz()
  if err!=nil {
     return nil, err
  }
  // do something with fl
// ...
}

// level 3
func baz() (*Fleazil, error) {
  If <disk is full> {
    return nil, error
  }
  // do something else....
}
```

So it should be clear to see that if the bottom level,`baz` discovers the disk is full, the error message about it will filter up to level 2 which will send it level 1, which will send it to whoever called `foo()`.